code/to_bsdd.py

```diff
@@ -171,28 +172,32 @@ def get_name(n):
 def reduce_description(s, trim=True):
     """ Remove all the unnecesarry characters, strip <html sections>, and trim the description to bare minimum. """
     try:
-        s1 = html.unescape(s)
+        s = html.unescape(s)
     except TypeError:
         # error when name contains link, for example: "https://github.com/buildingSMART/IFC4.3.x-development/edit/master/docs/schemas/core/IfcProductExtension/Types/IfcAlignmentTypeEnum.md#L0 has no content"
-        s1 = ''
+        s = ''
     if trim:
-        # split and remove on keywords:
-        i = min([x for x in [s1.find('NOTE'), s1.find('DIAGRAM'), s1.find('CHANGE'), s1.find('IFC4'), s1.find('HISTORY'), s1.find('REFERENCE'), s1.find('EXAMPLE'), s1.find('DEPRECATION'), 1e5] if x >= 0])
+        # split and remove on keywords:        
+        j = re.search(r'\$\$', s).start() if re.search(r'\$\$', s) else -1
+        k = re.search(r'(Base|General) formula', s).start() if re.search(r'(Base|General) formula', s) else -1
+        i = min([x for x in [s.find('NOTE'), s.find('DIAGRAM'), s.find('CHANGE'), s.find('IFC4'), s.find('HISTORY'), s.find('REFERENCE'), s.find('EXAMPLE'), s.find('DEPRECATION'), j, k, 1e5] if x >= 0])
         if i > 0 and i != 1e5:
-            s1 = s1[:i]
+            s = s[:i]
         elif i==0:
             # some descriptions start with History or Note. Don't want them.
-            s1 = ""
-    s2 = re.sub('\n', '; ', s1)
```

Comment:
You might want to do something like:

```python
replacements = [(re.compile(':'), ': '), (re.compile('\s+'), ' ')]
...
def reduce_description(s, trim=True):
    ...
    for pat, subs in replacements:
        s = re.sub(pat, subs, s)
```

This function gets executed a lot, precompiling the regexes once outside of the function might shave off some milliseconds. Writing it as a loop is less lines of code and less opportunities to make copy paste mistakes.